### PR TITLE
Run pack's validations on marketplaces change

### PR DESCRIPTION
**Pull request by DinaMeylakh**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7189

## Description
Run pack's validations on marketplaces change
Was checked locally:
Run on a pack_metadata change that was not marketplaces: not validated pack's files
Run on a pack_metadata change that was marketplaces: validated all pack's files
Run on a pack_metadata change and integration change: validated all pack's files and the integration only once.

### CODE REVIEW

File: demisto_sdk/commands/validate/validate_manager.py [hunk 1]
```diff
@@ -581,15 +582,19 @@ def validate_packs(
 
         return all(all_packs_valid)
 
-    def run_validations_on_pack(self, pack_path):
+    def run_validations_on_pack(self, pack_path, skip_files=None):
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validate_manager.py:585]:
```suggestion
    def run_validations_on_pack(self, pack_path, skip_files: Optional[Set[str]]=None):
```

**Review by YuvHayun — APPROVED** [review]
Very nice!

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 2]
```diff
@@ -267,6 +267,21 @@ def _parse_file_into_list(self, file_name, delimiter="\n"):
 
         return False
 
+    def check_metadata_for_marketplace_change(self):
+        """Return True if pack_metadata's marketplaces field was changed."""
+        metadata_file_path = self._get_pack_file_path(self.pack_meta_file)
+        if not os.path.isfile(metadata_file_path):
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:273]:
```suggestion
        metadata_file_path = self._get_pack_file_path(self.pack_meta_file)
        if not Path(metadata_file_path).is_file():
```

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 3]
```diff
@@ -267,6 +267,21 @@ def _parse_file_into_list(self, file_name, delimiter="\n"):
...
+            # No metadata file, No marketplace change.
+            return False
+
+        old_meta_file_content = get_remote_file(metadata_file_path, tag=self.prev_ver)
+        current_meta_file_content = self._read_metadata_content()
+        old_mps = old_meta_file_content.get("marketplaces", [])
+        current_mps = current_meta_file_content.get("marketplaces", [])
+        if set(old_mps) == set(current_mps):
+            return False
+        return True
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:283]:
```suggestion
        return set(old_mps) != set(current_mps):
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 * Calling **format** with the `-d` flag now removes test playbooks testing the deprecated content from conf.json.
 * Fixed an issue where in some cases the **split** command did not remove pack version note from the script.
 * Fixed an issue where **validate** would not properly detect dependencies of core packs.
+* Fixed an issue where **validate** did not run when a pack's marketplaces were changed.
```
[the remark this answers is not in the document]
Comment (YuvHayun) [reply CHANGELOG.md:6]: Agree

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 5]
```diff
@@ -267,6 +267,21 @@ def _parse_file_into_list(self, file_name, delimiter="\n"):
...
+        """Return True if pack_metadata's marketplaces field was changed."""
+        metadata_file_path = self._get_pack_file_path(self.pack_meta_file)
+        if not os.path.isfile(metadata_file_path):
+            # No metadata file, No marketplace change.
+            return False
+
+        old_meta_file_content = get_remote_file(metadata_file_path, tag=self.prev_ver)
+        current_meta_file_content = self._read_metadata_content()
+        old_mps = old_meta_file_content.get("marketplaces", [])
+        current_mps = current_meta_file_content.get("marketplaces", [])
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:280]:
```suggestion
        old_marketplaces = old_meta_file_content.get("marketplaces", [])
        current_marketplaces = current_meta_file_content.get("marketplaces", [])
```
let's make it easier on future us :)

**Pull request merged**